libs/d3d12core/main.c

```diff
@@ -53,6 +54,197 @@ static void *vulkan_module = NULL;
 #endif
 static PFN_vkGetInstanceProcAddr vulkan_vkGetInstanceProcAddr = NULL;
 
+#ifdef _WIN32
+static BOOL wait_vr_key(HKEY vr_key)
+{
+    DWORD type, value, wait_status, size;
+    DWORD max_retry = 10; /* Each value query and each timeout counts as a try. */
+    LSTATUS status;
+    HANDLE event;
+
+    size = sizeof(value);
+    if ((status = RegQueryValueExA(vr_key, "state", NULL, &type, (BYTE *)&value, &size)))
+    {
+        ERR("OpenVR: could not query value, status %d", status);
+        return false;
+    }
+    if (type != REG_DWORD)
+    {
+        ERR("OpenVR: unexpected value type %d", type);
+        return false;
+    }
+
+    if (value)
+        return value == 1;
+
+    event = CreateEventA(NULL, FALSE, FALSE, NULL);
+    if (event == NULL)
+    {
+        ERR("Cannot create event");
+        return false;
+    }
+
+    while (max_retry)
+    {
+        if (RegNotifyChangeKeyValue(vr_key, FALSE, REG_NOTIFY_CHANGE_LAST_SET, event, TRUE))
+        {
+            ERR("Error registering registry change notification");
+            break;
+        }
+        size = sizeof(value);
+        if ((status = RegQueryValueExA(vr_key, "state", NULL, &type, (BYTE *)&value, &size)))
+        {
+            ERR("OpenVR: could not query value, status %d", status);
+            break;
+        }
+        if (value)
+            break;
+        max_retry--;
+
+        while ((wait_status = WaitForSingleObject(event, 1000)) == WAIT_TIMEOUT && max_retry)
+        {
+            WARN("VR state wait timeout (retries left %u)", max_retry);
+            max_retry--;
+        }
+
+        if (wait_status != WAIT_OBJECT_0)
+        {
+            ERR("Got unexpected wait status ", wait_status);
+            break;
+        }
+    }
+
+    CloseHandle(event);
+    return value == 1;
+}
+
+static char *openvr_instance_extensions()
+{
+    HKEY vr_key;
+
+    LSTATUS status = RegOpenKeyA(HKEY_CURRENT_USER, "Software\\Wine\\VR", &vr_key);
+    char *openvr_instance_extensions = NULL;
+    DWORD len = 0, type;
+
+    if (status != ERROR_SUCCESS)
+    {
+        WARN("Failed to open VR registry key, status %d.\n", status);
```

Comment:
We expect this path to be taken if vkd3d-proton is running outside of Steam, right?

Reply:
Yeah